src/Input.js

```diff
@@ -27,16 +28,13 @@ const Input = ({
   const input = <input {...other} className={className} />
   if (icon) {
     return (
-      <div className={cx('uk-form-icon', { 'uk-form-icon-flip': flip })}>
-        <i
-          className={cx(`uk-icon-${icon}`, {
-            'uk-icon-spin': icon === 'spinner' || icon === 'refresh',
-          })}
-        />
+      <div className="uk-inline">
+        {icon && <Icon className={cx('uk-form-icon', { 'uk-form-icon-flip': flip })} icon={icon} /> }
```

Comment:
There is already a check made on line 29 to see if `icon` is truthy, it's not necessary to check it again on line 32.